### ISSUE TITLE
The influence of a GPU card on a model's accuracy 

### ISSUE THREAD

**Issue by VanitarNordic** [NVIDIA/DIGITS#1431]
Hello,

I decided to buy a GTX 1060 or GTX 1070 card to try with Deep Learning, but I am curious if the RAM size of The GPU or its bandwidth/speed will affect the **accuracy** of the final model or not, by comparing these two specific GPU cards.

in the other word, I want to know selecting the GTX 1060 will just cause longer training time over GTX 1070, or it will affect the accuracy of the model either.

**Comment by lukeyeager** [comment]
If running the same problem on different hardware produces different results then there's a bug in somebody's code (i.e. cuDNN, the framework or DIGITS). I can't promise that won't happen, but I wouldn't worry about it.

However, if you run on a different GPU with more memory, then you'll probably choose to train with a larger batch size. If you change the batch size, then you're changing the problem and you probably WILL see slightly different results. Typically you want to increase the learning rate when you increase the batch size, but exactly how to do that is as yet unknown.

> Theory aside, for the batch sizes considered in this note, the heuristic that I found to work the best was to multiply the learning rate by k when multiplying the batch size by k. I can’t explain this discrepancy between theory and practice.
https://arxiv.org/pdf/1404.5997v2.pdf

... but that's probably more information than you wanted. Basically yes - if you get a faster GPU you'll converge faster to the same accuracy.

**Comment by VanitarNordic** [comment]
Thank you very much for your explanation.

Exactly the tricky area is adjusting the batch sizes in relation with the GPU memory size or bandwidth.

if I select the 1060 with 6G memory, the I have to adjust batch sizes for 6G,. for 1070 with 8G memory larger batch sizes can be applied. is this affect the **model accuracy**?
 (apart from that training with 1070 will be finished faster).

**Comment by lukeyeager** [comment]
As I said, changing the batch size will change the results slightly. My intuition is that training on a larger batch size should decrease overfitting to your training data (higher `train_loss` and lower `train_accuracy`). It's possible this would translate to a better generalization to your validation data (lower `val_loss` and higher `val_accuracy`) but I'm really not sure.

I wouldn't expect going from 6GB to 8GB would have a huge impact either way. If instead you were talking about going from a single 4GB card to eight 12GB cards, then this would be a much more important question.

**Comment by VanitarNordic** [comment]
Thank you very much gentleman. I got my answer.
